autounsuspend.lua

```diff
@@ -8,19 +8,35 @@ local persist = require('persist-table')
 local GLOBAL_KEY = 'autounsuspend' -- used for state change hooks and persistence
 
 enabled = enabled or false
+smart = smart or nil
+if smart == nil then
+    smart = true
+end
 
 function isEnabled()
     return enabled
 end
 
 local function persist_state()
-    persist.GlobalTable[GLOBAL_KEY] = json.encode({enabled=enabled})
+    persist.GlobalTable[GLOBAL_KEY] = json.encode({
+        enabled=enabled,
+        smart=smart
+    })
+end
+
+function set_smart_mode(enable_smart_mode)
+    smart = enable_smart_mode
+    persist_state()
 end
 
 local function event_loop()
     if enabled then
-        dfhack.run_script('unsuspend', '--quiet')
-        dfhack.timeout(1, 'days', event_loop)
+        local args = {'--quiet'}
+        if smart then
+            table.insert(args, '--smart')
+        end
+        dfhack.run_script('unsuspend', table.unpack(args))
+        dfhack.timeout(10, 'ticks', event_loop)
```

Comment:
yeah, this is *really* frequent, even more frequent than `autonestbox`, which has similar requirements for "mark it before someone takes it". I'd be concerned about running even cpp code at this frequency. Registering a job completion handler might be the better approach.

Reply:
I switched to triggering on job creation and completion, and it works as expected :)